src/utils/include/utils/LCA.hpp

```diff
@@ -167,18 +169,21 @@ inline int LCA::getLCA( int u, int v ) const
     return m_euler[queryRMQ( m_firstAppearance[u], m_firstAppearance[v] )];
 }
 
-// TODO: this function should be const! For that, we cannot use operator[] for the maps. They
-// are not const and can potentially modify the maps. We should maybe call m_encode.at(...).
-inline std::string LCA::getLCA( const std::vector< std::string >& taxIds )
+inline std::string LCA::getLCA( const std::vector< std::string >& taxIds ) const
 {
-    assert( taxIds.size() > 1 ); // I wonder if, aside from asserting this, if we should just branch off...
+    // check for valid entries
+    for ( auto it = taxIds.begin(); it != taxIds.end(); ++it )
```

Comment:
Looks like this could be:

```
if ( std::any_of( taxIds.begin(), taxIds.end(), [&]( const auto& id ) { return m_encode.count( id ) == 0; } ) )
{
    return "1";
}
```